WickedEngine/wiRenderer.cpp

```diff
@@ -9414,383 +9443,951 @@ void Postprocess_RTReflection(
 
 		const GPUResource* uavs[] = {
-			&output,
+			&res.buffer_tile_tracing_statistics,
+			&res.buffer_tiles_tracing_earlyexit,
+			&res.buffer_tiles_tracing_cheap,
+			&res.buffer_tiles_tracing_expensive,
+			&res.texture_tile_minmax_roughness,
 		};
 		device->BindUAVs(uavs, 0, arraysize(uavs), cmd);
 
 		{
 			GPUBarrier barriers[] = {
-				GPUBarrier::Image(&output, output.desc.layout, ResourceState::UNORDERED_ACCESS),
+				GPUBarrier::Image(&res.texture_tile_minmax_roughness, res.texture_tile_minmax_roughness.desc.layout, ResourceState::UNORDERED_ACCESS),
 			};
 			device->Barrier(barriers, arraysize(barriers), cmd);
 		}
 
 		device->Dispatch(
-			(desc.width + POSTPROCESS_BLOCKSIZE - 1) / POSTPROCESS_BLOCKSIZE,
-			(desc.height + POSTPROCESS_BLOCKSIZE - 1) / POSTPROCESS_BLOCKSIZE,
+			(res.texture_tile_minmax_roughness.GetDesc().width + POSTPROCESS_BLOCKSIZE - 1) / POSTPROCESS_BLOCKSIZE,
+			(res.texture_tile_minmax_roughness.GetDesc().height + POSTPROCESS_BLOCKSIZE - 1) / POSTPROCESS_BLOCKSIZE,
 			1,
 			cmd
 		);
 
 		{
 			GPUBarrier barriers[] = {
 				GPUBarrier::Memory(),
-				GPUBarrier::Image(&output, ResourceState::UNORDERED_ACCESS, output.desc.layout),
+				GPUBarrier::Image(&res.texture_tile_minmax_roughness, ResourceState::UNORDERED_ACCESS, res.texture_tile_minmax_roughness.desc.layout),
 			};
 			device->Barrier(barriers, arraysize(barriers), cmd);
 		}
 
 		device->EventEnd(cmd);
 	}
 
-	int temporal_output = device->GetFrameCount() % 2;
-	int temporal_history = 1 - temporal_output;
+	// Kick indirect tile jobs:
+	{
+		device->EventBegin("SSR Kickjobs", cmd);
+		device->BindComputeShader(&shaders[CSTYPE_POSTPROCESS_SSR_KICKJOBS], cmd);
 
-	// Temporal pass:
+		const GPUResource* uavs[] = {
+			&res.buffer_tile_tracing_statistics,
+			&res.buffer_tiles_tracing_earlyexit,
+			&res.buffer_tiles_tracing_cheap,
+			&res.buffer_tiles_tracing_expensive,
+		};
+		device->BindUAVs(uavs, 0, arraysize(uavs), cmd);
+
+		device->Dispatch(1, 1, 1, cmd);
+
+		GPUBarrier barriers[] = {
+			GPUBarrier::Memory(),
+			GPUBarrier::Buffer(&res.buffer_tile_tracing_statistics, ResourceState::UNORDERED_ACCESS, ResourceState::INDIRECT_ARGUMENT),
+		};
+		device->Barrier(barriers, arraysize(barriers), cmd);
+
+		device->EventEnd(cmd);
+	}
+
+	PostProcess postprocess;
+
+	// Depth hierarchy:
 	{
-		device->EventBegin("Temporal pass", cmd);
-		device->BindComputeShader(&shaders[CSTYPE_POSTPROCESS_SSR_TEMPORAL], cmd);
-		device->PushConstants(&postprocess, sizeof(postprocess), cmd);
+		device->EventBegin("SSR Depth hierarchy pass", cmd);
+		device->BindComputeShader(&shaders[CSTYPE_POSTPROCESS_SSR_DEPTHHIERARCHY], cmd);
+
+		TextureDesc hierarchyDesc = res.texture_depth_hierarchy.GetDesc();
 
-		device->BindResource(&output, 0, cmd);
-		device->BindResource(&res.texture_temporal[temporal_history], 1, cmd);
-		device->BindResource(&res.rayLengths, 3, cmd);
+		{
+			device->BindResource(&depthBuffer_Main, 0, cmd);
```

Comment:
The depthBuffer_Main no longer needs to be bound, it is available always globally in shaders as `texture_depth` (in globals.hlsli)

Reply:
(The idea is that commonly used textures and resources are now available with bindless system, and accessible from `GetCamera()`, or `GetScene()`, or `GetFrame()` in the shaders. For example the `texture_depth` is accessible from camera, but you can just use `texture_depth` macro too)

Reply:
You are right, I should've done that from the start. Just made a commit with your suggestion. I thought at the time of getting the rawest depth, but as you mentioned they're the same.